src/Http/Client/Auth/Digest.php

```diff
@@ -26,13 +27,61 @@
  */
 class Digest
 {
+    /**
+     * Algorithms
+     */
+    public const MD5 = 'MD5';
+    public const SHA_256 = 'SHA-256';
+    public const SHA_512_256 = 'SHA-512-256';
+    public const MD5_SESS = 'MD5-sess';
+    public const SHA_256_SESS = 'SHA-256-sess';
+    public const SHA_512_256_SESS = 'SHA-512-256-sess';
+
+    /**
+     * QOP
+     */
+    public const QOP_AUTH = 'auth';
+    public const QOP_AUTH_INT = 'auth-int';
+
+    /**
+     * Algorithms <-> Hash type
+     */
+    public const ALGORITHMS = [
+        self::MD5 => 'md5',
```

Comment:
This looks confusing. Why not make the algorithm constants the same?

Reply:
Basically the array maps Digest Algorithms and Hash Algorithm which at the end of the day are similar but not the same. I was thinking about it and found this way but if you have another suggestion I'm open to change it of course.

Reply:
I think I'm just confused by the names since the array is called ALGORITHMS and the constants are all algorithms. 

We should probably prefix the constants with ALGO_ and name the array HASHES or something like that.

Reply:
So do rename those constants or what should be done here?

Reply:
Yeah, I am just updating the PR with the change

Reply:
That's my suggestion to make it clearer.

Reply:
Did it already some time ago, please take a look and let me know if anything else is needed. Build is failing because for some reason coverage decreased 0.02% after renaming constants 🤷🏻‍♂️

Reply:
I think it looks much better. If @LordSimal is still good with the rest, then I have no objections.

Reply:
Sorry, missed that one. Just tested the latest version and everything works fine. So from my side this can be merged 👍🏻